src/csr.tex

```diff
@@ -194,16 +198,19 @@ \subsection*{CSR Access Ordering}
 the CSRs defined thus far in Volumes I and II of this specification are not
 directly accessible to other harts or devices and cause no side effects visible
 to other harts or devices.  Thus, accesses to CSRs other than the
-aforementioned three can be freely reordered with respect to FENCE instructions
+aforementioned three can be freely reordered in the global memory order
+with respect to FENCE instructions
 without violating this specification.
 \end{commentary}
 
-For CSR accesses that cause side effects, the above ordering constraints apply
-to the order of the initiation of those side effects but does not necessarily
-apply to the order of the completion of those side effects.
-
 The hardware platform may define that accesses to certain CSRs are
 strongly ordered, as defined by the Memory-Ordering PMAs section in Volume II
 of this manual. Accesses to strongly ordered CSRs have stronger ordering
 constraints with respect to accesses to both weakly ordered CSRs and accesses
 to memory-mapped I/O regions.
+
+\begin{commentary}
+The rules for the reordering of CSR accesses in the global memory order
+should probably be moved to Chapter~\ref{ch:memorymodel} concerning the
+RVWMO memory consistency model.
+\end{commentary}
```

Comment:
With regards to the interaction between the memory ordering model and the CSR access ordering model, it seems appropriate to discuss those behaviors in the RVWMO section.

Generally speaking, though, I think this section belongs in a broader "instruction execution" model, that ties together all accesses and modifications of architectural state, including implicit memory accesses (e.g. instruction fetches and table walks), explicit memory accesses, CSR accesses, traps, etc. (In general, RV seems to require a pretty strict "sequential execution model," but there are exceptions; otherwise, there wouldn't be FENCE.I or SFENCE.VMA....)

RVWMO only captures a portion of the interesting events, so perhaps that would be expanded or incorporated into a broader conceptual model some day. (Or we just wait for the formal model....?)

Reply:
I agree there is still much to be done to standardize and document these matters, just as you say.  The comment I've added is merely a temporary patch, obviously.